rclpy/rclpy/executors.py

```diff
@@ -366,7 +366,19 @@ def spin_once_until_future_complete(
     def _take_timer(self, tmr):
         try:
             with tmr.handle:
-                tmr.handle.call_timer()
+                info = tmr.handle.call_timer_with_info()
```

Comment:
This still works but,

something we can do for the performance improvement is to get the signature when we create timer, not calling here every single time, and call either `call_timer()` or `call_timer_with_info()` accordingly?

Reply:
in the rcl layer, there is none.

Reply:
@mjcarroll sorry i was not clear on this.

i was thinking that calling `inspect.signature` could be optimized to cache the callback type or something, but i guess that approach is bad assumption since user application would change the callback for `Timer` class. besides, this is gonna be really minor optimization. i will keep the current code for now.